File: FR/java/src/com/arcsoft/AFR_FSDK_Version.java
package com.arcsoft;

import java.util.Arrays;
import java.util.List;

import com.sun.jna.Structure;

public class AFR_FSDK_Version extends Structure {
    public int lCodebase;
    public int lMajor;
    public int lMinor;
    public int lBuild;
    public int lFeatureLevel;
    public String Version;
    public String BuildDate;
    public String CopyRight;
    
    @Override
    protected List getFieldOrder() { 
        return Arrays.asList(new String[] { 
            "lCodebase", "lMajor", "lMinor", "lBuild","lFeatureLevel","Version", "BuildDate", "CopyRight"
        });
    }
}

File: FR/C#/AFR_FSDK_Version.cs
using System.Runtime.InteropServices;

namespace arcsoft {
    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
    public struct AFR_FSDK_Version {
        public int lCodebase;
        public int lMajor;
        public int lMinor;
        public int lBuild;
        public int lFeatureLevel;
        public string Version;
        public string BuildDate;
        public string CopyRight;
    };
}
